src/app/plugins/cat.plugin.ts

```diff
@@ -0,0 +1,143 @@
+  public usage: string = 'cat breed or pic bomb(optional) <breed (optional)>';
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Pets;
+
+  private _API_URL: string = 'https://api.thecatapi.com/v1/';
+  private _breeds: string[] = [];
+  private _breedsID: string[] = [];
+
+  constructor(public container: IContainer) {
+    super();
+    this.container.httpService
+      .get(`${this._API_URL}breeds`)
+      .then((response: IHttpResponse) => {
+        const breeds = response.data;
+
+        for (let index = 0; index < breeds.length; index++) {
+            const breed = breeds[index].name;
+            const breedID = breeds[index].id;
+            this._breeds.push(breed.toLowerCase())
+            this._breedsID.push(breedID.toLowerCase())
+        }
+    
+        // Doesn't work due to the the way the JSON info is now listed, Would be interested in how to do it with foreach if anyone could tell me.
+        // breeds.forEach((breed) => {
+        //   this._breeds.push(breed.toLowerCase());
+        // });
+      })
+      .catch((err) => console.log(err));
+  }
+
+  public async execute(message: IMessage, args?: string[]){
+
+    var command = this._parseCommand(args || []);
+    
+    if (command.includes("fact")) {
+
+        // fact is not yet supported by the API at this time however this code should work upon its release
+        
+        // command = command.replace("fact", "");
+        // if (command.includes("breeds")) {
+
+        //     message.reply(`Breeds supported: \n\`\`\`\n${this._breeds.join('\n')}\`\`\``);
+        //     return;
+
+        // } else {
+
+        //     var search = "";
+
+        //     if (command != "fact") {
+
+        //         search = command.replace("fact", "");
+
+
+        //         const index = this._breeds.findIndex((curBreed) => {
+        //             return command.includes(curBreed);
+        //         })
+
+        //         breed = `breed_id=${this._breedsID[index]}`;
+        //     }
+
+        //     await this.container.httpService
+        //     .get(`${this._API_URL}facts/${search}`)
+        //     .then((response: IHttpResponse) => {
+
+        //         message.reply('', {
+        //             files: [response.data[0].text],
+        //         });
+        //     })
+
+        //     .catch((err) => console.log(err));
+
+        // }
+
+    } else if (command == "breeds" || command == "breed") {
+        
+        //Simply return the list of suported breeds
+        message.reply(`Breeds supported: \n\`\`\`\n${this._breeds.join('\n')}\`\`\``);
+        return;
+
+    } else if (command == "breads" || command == "bread") {
+
+        // the Easter Egg, my gawd
+        message.reply('', {
+            files: [`https://www.thetoychronicle.com/wp-content/uploads/2016/05/Box-cat-Bread-cat-Rato-workroom-Kim-.jpg`],
+        });
+
+    } else if (command.includes("pic")){
```

Comment:
`.includes()` should really be some prefix match right?

Reply:
I personally prefer the includes, as this allows the user to ignore order.  Although if you want it changing to that I can change it to that.